SportsGoods.App/Services/ProductService.cs

```diff
@@ -0,0 +1,75 @@
+﻿using SportsGoods.Core.Interfaces;
+using SportsGoods.Core.Models;
+using SportsGoods.Data.DAL;
+using System.Xml.Linq;
+
+namespace SportsGoods.App.Services
+{
+    public class ProductService
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IProductRepository _productRepository;
+
+        public ProductService(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+        public ProductService(ApplicationDbContext context,IProductRepository productRepository)
+        {
+            _context = context;
+            _productRepository = productRepository;
+        }
+
+        public async Task SeedProductsFromXmlAsync(string xmlFilePath)
+        {
+            XDocument doc = XDocument.Load(xmlFilePath);
+
+            var products = new List<Product>();
+            foreach (var element in doc.Descendants("Product"))
+            {
+                var idElement = element.Element("Id");
+                var titleElement = element.Element("Title");
+                var descriptionElement = element.Element("Description");
+                var brandElement = element.Element("Brand");
+                var priceElement = element.Element("Price");
+                var quantityElement = element.Element("Quantity");
+                var productCategoryElement = element.Element("ProductCategory");
+
+                Guid productId;
+
+                if (idElement != null && Guid.TryParse(idElement.Value, out productId))
+                {
+                    if (!string.IsNullOrEmpty(titleElement?.Value) &&
```

Comment:
Extract validation in a separate method to improve readability @LVarsaminov

Reply:
Done